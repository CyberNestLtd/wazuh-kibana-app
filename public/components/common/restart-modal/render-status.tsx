/*
 * Wazuh app - React component for render status nodes.
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import { RestartHandler } from '../../../react-services/wz-restart';
import {
  EuiDescriptionListTitle,
  EuiDescriptionListDescription,
  EuiIconTip,
  EuiLoadingSpinner,
} from '@elastic/eui';

export const RenderStatus = (props) => {
  const { node, statusRestart } = props;

  switch (statusRestart) {
    case RestartHandler.RESTART_STATES.SYNCING:
      return (
        <>
          <EuiDescriptionListTitle>{node.name}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription className="wz-text-left">
            {node.synced ? (
              <EuiIconTip aria-label="Success" size="m" type="check" color="success" />
            ) : (
              <EuiLoadingSpinner size="m" />
            )}
          </EuiDescriptionListDescription>
        </>
      );
    case RestartHandler.RESTART_STATES.SYNC_ERROR:
      return (
        <>
          <EuiDescriptionListTitle>{node.name}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription className="wz-text-left">
            {node.synced ? (
              <EuiIconTip aria-label="Synchronized" size="m" type="check" color="success" />
            ) : (
              <EuiIconTip aria-label="Unsynchronized" size="m" type="alert" color="danger" />
            )}
          </EuiDescriptionListDescription>
        </>
      );
    case RestartHandler.RESTART_STATES.RESTARTING:
      return (
        <>
          <EuiDescriptionListTitle>{node.node}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription className="wz-text-left">
            {node.isRestarted ? (
              <EuiIconTip aria-label="Synchronized" size="m" type="check" color="success" />
            ) : (
              <EuiLoadingSpinner size="m" />
            )}
          </EuiDescriptionListDescription>
        </>
      );
    case RestartHandler.RESTART_STATES.RESTART_ERROR:
      return (
        <>
          <EuiDescriptionListTitle>{node.node}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription className="wz-text-left">
            {node.isRestarted ? (
              <EuiIconTip aria-label="Synchronized" size="m" type="check" color="success" />
            ) : (
              <EuiIconTip aria-label="Unsynchronized" size="m" type="alert" color="danger" />
            )}
          </EuiDescriptionListDescription>
        </>
      );
      case RestartHandler.RESTART_STATES.RESTARTED_INFO:
      return (
        <>
          <EuiDescriptionListTitle>{node.node}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription className="wz-text-left">
            <EuiIconTip aria-label="Synchronized" size="m" type="check" color="success" />
          </EuiDescriptionListDescription>
        </>
      );
    default:
      return (
        <>
          <EuiDescriptionListTitle>{node.name}</EuiDescriptionListTitle>
          <EuiDescriptionListDescription>
            {node.synced ? (
              <EuiIconTip aria-label="Success" size="m" type="check" color="success" />
            ) : (
              <EuiLoadingSpinner size="m" />
            )}
          </EuiDescriptionListDescription>
        </>
      );
  }
};
