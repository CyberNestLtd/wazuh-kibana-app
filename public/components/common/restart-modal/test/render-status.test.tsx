/*
 * Wazuh app - React test for table default component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import { mount } from 'enzyme';
import { RenderStatus } from '../render-status';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const restartingState = {
  node: { node: 'nodo-master', isRestarted: false },

  statusRestart: 'restarting',
};
const restartedInfoState = {
  node: { node: 'nodo-master', isRestarted: true },

  statusRestart: 'restarted_info',
};
const restartErrorState = {
  node: { node: 'nodo-master', isRestarted: true },

  statusRestart: 'restart_error',
};
const syncingState = {
  node: { node: 'nodo-master', synced: true },
  statusRestart: 'syncing',
};
const syncErrorState = {
  node: { node: 'nodo-master', synced: true },

  statusRestart: 'sync_error',
};
const mockStore = configureMockStore();

describe('Restart status: restarting', () => {
  it('renders correctly to match the snapshot.', () => {
    const store = mockStore(restartingState);

    const wrapper = mount(
      <Provider store={store}>
        <RenderStatus node={restartingState.node} statusRestart={restartingState.statusRestart} />;
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Restart status: restarted_info', () => {
  it('renders correctly to match the snapshot.', () => {
    const store = mockStore(restartedInfoState);

    const wrapper = mount(
      <Provider store={store}>
        <RenderStatus
          node={restartedInfoState.node}
          statusRestart={restartedInfoState.statusRestart}
        />
        ;
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Restart status: restart_error', () => {
  it('renders correctly to match the snapshot.', () => {
    const store = mockStore(restartErrorState);

    const wrapper = mount(
      <Provider store={store}>
        <RenderStatus
          node={restartErrorState.node}
          statusRestart={restartErrorState.statusRestart}
        />
        ;
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Restart status: sync_error', () => {
  it('renders correctly to match the snapshot.', () => {
    const store = mockStore(syncErrorState);

    const wrapper = mount(
      <Provider store={store}>
        <RenderStatus node={syncErrorState.node} statusRestart={syncErrorState.statusRestart} />;
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Restart status: syncing', () => {
  it('renders correctly to match the snapshot.', () => {
    const store = mockStore(syncingState);
    const cancelModal = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <RenderStatus node={syncingState.node} statusRestart={syncingState.statusRestart} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
