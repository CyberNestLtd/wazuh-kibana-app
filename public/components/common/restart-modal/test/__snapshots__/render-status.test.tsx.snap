// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restart status: restart_error renders correctly to match the snapshot. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RenderStatus
    node={
      Object {
        "isRestarted": true,
        "node": "nodo-master",
      }
    }
    statusRestart="restart_error"
  >
    <EuiDescriptionListTitle>
      <dt
        className="euiDescriptionList__title"
      >
        <span
          className="euiToolTipAnchor"
        >
          <WzTextWithTooltipIfTruncated
            elementStyle={
              Object {
                "maxWidth": "85px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "block",
                  "maxWidth": "85px",
                  "overflow": "hidden",
                  "paddingBottom": "3px",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </WzTextWithTooltipIfTruncated>
        </span>
      </dt>
    </EuiDescriptionListTitle>
    <EuiDescriptionListDescription
      className="wz-text-left"
    >
      <dd
        className="euiDescriptionList__description wz-text-left"
      >
        <EuiIconTip
          aria-label="Synchronized"
          color="success"
          size="m"
          type="check"
        >
          <EuiToolTip
            delay="regular"
            position="top"
          >
            <span
              className="euiToolTipAnchor"
              onKeyUp={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <EuiIcon
                aria-label="Synchronized"
                color="success"
                onBlur={[Function]}
                onFocus={[Function]}
                size="m"
                tabIndex={0}
                type="check"
              >
                <EuiIconCheck
                  aria-label="Synchronized"
                  className="euiIcon euiIcon--medium euiIcon--success"
                  focusable="true"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="img"
                  style={null}
                  tabIndex={0}
                >
                  <svg
                    aria-label="Synchronized"
                    className="euiIcon euiIcon--medium euiIcon--success"
                    focusable="true"
                    height={16}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="img"
                    style={null}
                    tabIndex={0}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.5 12a.502.502 0 01-.354-.146l-4-4a.502.502 0 01.708-.708L6.5 10.793l6.646-6.647a.502.502 0 01.708.708l-7 7A.502.502 0 016.5 12"
                      fillRule="evenodd"
                    />
                  </svg>
                </EuiIconCheck>
              </EuiIcon>
            </span>
          </EuiToolTip>
        </EuiIconTip>
      </dd>
    </EuiDescriptionListDescription>
  </RenderStatus>
  ;
</Provider>
`;

exports[`Restart status: restarted_info renders correctly to match the snapshot. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RenderStatus
    node={
      Object {
        "isRestarted": true,
        "node": "nodo-master",
      }
    }
    statusRestart="restarted_info"
  >
    <EuiDescriptionListTitle>
      <dt
        className="euiDescriptionList__title"
      >
        <span
          className="euiToolTipAnchor"
        >
          <WzTextWithTooltipIfTruncated
            elementStyle={
              Object {
                "maxWidth": "85px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "block",
                  "maxWidth": "85px",
                  "overflow": "hidden",
                  "paddingBottom": "3px",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </WzTextWithTooltipIfTruncated>
        </span>
      </dt>
    </EuiDescriptionListTitle>
    <EuiDescriptionListDescription
      className="wz-text-left"
    >
      <dd
        className="euiDescriptionList__description wz-text-left"
      >
        <EuiIconTip
          aria-label="Synchronized"
          color="success"
          size="m"
          type="check"
        >
          <EuiToolTip
            delay="regular"
            position="top"
          >
            <span
              className="euiToolTipAnchor"
              onKeyUp={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <EuiIcon
                aria-label="Synchronized"
                color="success"
                onBlur={[Function]}
                onFocus={[Function]}
                size="m"
                tabIndex={0}
                type="check"
              >
                <EuiIconEmpty
                  aria-hidden={true}
                  aria-label="Synchronized"
                  className="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
                  focusable="true"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="img"
                  style={null}
                  tabIndex={0}
                >
                  <svg
                    aria-hidden={true}
                    aria-label="Synchronized"
                    className="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
                    focusable="true"
                    height={16}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="img"
                    style={null}
                    tabIndex={0}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </EuiIconEmpty>
              </EuiIcon>
            </span>
          </EuiToolTip>
        </EuiIconTip>
      </dd>
    </EuiDescriptionListDescription>
  </RenderStatus>
  ;
</Provider>
`;

exports[`Restart status: restarting renders correctly to match the snapshot. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RenderStatus
    node={
      Object {
        "isRestarted": false,
        "node": "nodo-master",
      }
    }
    statusRestart="restarting"
  >
    <EuiDescriptionListTitle>
      <dt
        className="euiDescriptionList__title"
      >
        <span
          className="euiToolTipAnchor"
        >
          <WzTextWithTooltipIfTruncated
            elementStyle={
              Object {
                "maxWidth": "85px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "block",
                  "maxWidth": "85px",
                  "overflow": "hidden",
                  "paddingBottom": "3px",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </WzTextWithTooltipIfTruncated>
        </span>
      </dt>
    </EuiDescriptionListTitle>
    <EuiDescriptionListDescription
      className="wz-text-left"
    >
      <dd
        className="euiDescriptionList__description wz-text-left"
      >
        <EuiLoadingSpinner
          size="m"
        >
          <span
            className="euiLoadingSpinner euiLoadingSpinner--medium"
          />
        </EuiLoadingSpinner>
      </dd>
    </EuiDescriptionListDescription>
  </RenderStatus>
  ;
</Provider>
`;

exports[`Restart status: sync_error renders correctly to match the snapshot. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RenderStatus
    node={
      Object {
        "node": "nodo-master",
        "synced": true,
      }
    }
    statusRestart="sync_error"
  >
    <EuiDescriptionListTitle>
      <dt
        className="euiDescriptionList__title"
      >
        <span
          className="euiToolTipAnchor"
        >
          <WzTextWithTooltipIfTruncated
            elementStyle={
              Object {
                "maxWidth": "85px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "block",
                  "maxWidth": "85px",
                  "overflow": "hidden",
                  "paddingBottom": "3px",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </WzTextWithTooltipIfTruncated>
        </span>
      </dt>
    </EuiDescriptionListTitle>
    <EuiDescriptionListDescription
      className="wz-text-left"
    >
      <dd
        className="euiDescriptionList__description wz-text-left"
      >
        <EuiIconTip
          aria-label="Synchronized"
          color="success"
          size="m"
          type="check"
        >
          <EuiToolTip
            delay="regular"
            position="top"
          >
            <span
              className="euiToolTipAnchor"
              onKeyUp={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <EuiIcon
                aria-label="Synchronized"
                color="success"
                onBlur={[Function]}
                onFocus={[Function]}
                size="m"
                tabIndex={0}
                type="check"
              >
                <EuiIconCheck
                  aria-label="Synchronized"
                  className="euiIcon euiIcon--medium euiIcon--success"
                  focusable="true"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="img"
                  style={null}
                  tabIndex={0}
                >
                  <svg
                    aria-label="Synchronized"
                    className="euiIcon euiIcon--medium euiIcon--success"
                    focusable="true"
                    height={16}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="img"
                    style={null}
                    tabIndex={0}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.5 12a.502.502 0 01-.354-.146l-4-4a.502.502 0 01.708-.708L6.5 10.793l6.646-6.647a.502.502 0 01.708.708l-7 7A.502.502 0 016.5 12"
                      fillRule="evenodd"
                    />
                  </svg>
                </EuiIconCheck>
              </EuiIcon>
            </span>
          </EuiToolTip>
        </EuiIconTip>
      </dd>
    </EuiDescriptionListDescription>
  </RenderStatus>
  ;
</Provider>
`;

exports[`Restart status: syncing renders correctly to match the snapshot. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RenderStatus
    node={
      Object {
        "node": "nodo-master",
        "synced": true,
      }
    }
    statusRestart="syncing"
  >
    <EuiDescriptionListTitle>
      <dt
        className="euiDescriptionList__title"
      >
        <span
          className="euiToolTipAnchor"
        >
          <WzTextWithTooltipIfTruncated
            elementStyle={
              Object {
                "maxWidth": "85px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "block",
                  "maxWidth": "85px",
                  "overflow": "hidden",
                  "paddingBottom": "3px",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </WzTextWithTooltipIfTruncated>
        </span>
      </dt>
    </EuiDescriptionListTitle>
    <EuiDescriptionListDescription
      className="wz-text-left"
    >
      <dd
        className="euiDescriptionList__description wz-text-left"
      >
        <EuiIconTip
          aria-label="Success"
          color="success"
          size="m"
          type="check"
        >
          <EuiToolTip
            delay="regular"
            position="top"
          >
            <span
              className="euiToolTipAnchor"
              onKeyUp={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <EuiIcon
                aria-label="Success"
                color="success"
                onBlur={[Function]}
                onFocus={[Function]}
                size="m"
                tabIndex={0}
                type="check"
              >
                <EuiIconCheck
                  aria-label="Success"
                  className="euiIcon euiIcon--medium euiIcon--success"
                  focusable="true"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  role="img"
                  style={null}
                  tabIndex={0}
                >
                  <svg
                    aria-label="Success"
                    className="euiIcon euiIcon--medium euiIcon--success"
                    focusable="true"
                    height={16}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="img"
                    style={null}
                    tabIndex={0}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.5 12a.502.502 0 01-.354-.146l-4-4a.502.502 0 01.708-.708L6.5 10.793l6.646-6.647a.502.502 0 01.708.708l-7 7A.502.502 0 016.5 12"
                      fillRule="evenodd"
                    />
                  </svg>
                </EuiIconCheck>
              </EuiIcon>
            </span>
          </EuiToolTip>
        </EuiIconTip>
      </dd>
    </EuiDescriptionListDescription>
  </RenderStatus>
</Provider>
`;
